app/src/main/java/memphis/myapplication/UI/AddFriendFragment.java

```diff
@@ -0,0 +1,345 @@
+    private Toolbar toolbar;
+    private String friendName;
+    private String friendDomain;
+    private View addFriendView;
+    private int nextState = 0;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        addFriendView = inflater.inflate(R.layout.fragment_add_friend, container, false);
+
+        databaseViewModel = ViewModelProviders.of(getActivity()).get(RealmViewModel.class);
+
+        UserModel userModel = ViewModelProviders.of(getActivity(), new ViewModelProvider.Factory() {
+            @NonNull
+            @Override
+            public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
+                return (T) new UserModel(getActivity());
+            }
+        }).get(UserModel.class);
+
+        setupToolbar(userModel.getUserImage().getValue());
+        setButtonWidth();
+
+        setListeners();
+
+        if (nextState != NONE){
+            if (nextState == DISPLAY_QR){
+                displayMyQR(new View(getActivity()));
+                nextState = NONE;
+            }else{
+                scanFriendQR(new View(getActivity()));
+                nextState = NONE;
+            }
+        }
+        return addFriendView;
+    }
+
+    private void setListeners() {
+        addFriendView.findViewById(R.id.scanFriendButton).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                scanFriendQR(v);
+            }
+        });
+
+        addFriendView.findViewById(R.id.showYourCode).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                displayMyQR(v);
+            }
+        });
+
+        addFriendView.findViewById(R.id.addFriend).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+                builder.setTitle("Display or scan QR code first?").setCancelable(false);
+                final View tempView = view;
+
+                builder.setPositiveButton("Display QR code", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        // Send them to a new page to select friends to send photo to
+                        scanFriendQR(tempView);
+                        nextState = DISPLAY_QR;
+//                        displayMyQR(tempView);
+
+                    }
+                });
+                builder.setNegativeButton("Scan QR code", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        displayMyQR(tempView);
+                        nextState = SCAN_QR;
+//                        scanFriendQR(tempView);
+
```

Comment:
These are reversed. Pressing "Display QR code" brings up the scanning screen and pressing "Scan QR code" brings up the QR code.